metadata-ingestion/src/datahub/utilities/source_helpers.py

```diff
@@ -148,3 +155,44 @@ def auto_materialize_referenced_tags(
             entityUrn=urn,
             aspect=TagKeyClass(name=tag_urn.get_entity_id()[0]),
         ).as_workunit()
+
+
+def auto_browse_path_v2(
+    stream: Iterable[MetadataWorkUnit],
+) -> Iterable[MetadataWorkUnit]:
+    """Generate BrowsePathsV2 from Container and BrowsePaths aspects."""
+
+    container_urns = set()
+    parents = {}
+    children: Dict[str, List[str]] = defaultdict(list)
+    for wu in stream:
+        yield wu
+
+        urn = wu.get_urn()
+        if guess_entity_type(urn) == "container":
+            container_urns.add(urn)
+
+        container_aspects = wu.get_aspects_of_type(ContainerClass)
+        for aspect in container_aspects:
+            parent = aspect.container
+            parents[urn] = parent
+            children[parent].append(urn)
+
+    # Yield browse paths v2 in topological order
+    paths: Dict[str, List[str]] = {}  # Maps urn -> list of urns in path
+    nodes = container_urns - parents.keys()  # Start with root containers
+    while nodes:
+        node = nodes.pop()
+        nodes.update(children[node])
+
+        if node not in parents:  # root
+            paths[node] = []
+        else:
+            parent = parents[node]
+            paths[node] = [*paths[parent], parent]
+        yield MetadataChangeProposalWrapper(
+            entityUrn=node,
+            aspect=BrowsePathsV2Class(
```

Comment:
We should also keep track of any BrowsePathsV2Class that we've already emitted. That way, if the source is already emitting BrowsePathsV2Class then we don't add a second one

Reply:
I was thinking once we implement creating them outright for a source we'd just remove the helper, but I could see this being useful in case we forget or if we're only doing partial creation manually

Reply:
Added